kernel/private/classes/ezpkernelweb.php

```diff
@@ -843,15 +843,10 @@ protected function dispatchLoop()
                 $moduleResult = $this->module->handleError( eZError::KERNEL_MODULE_DISABLED, 'kernel', array( 'check' => $moduleCheck ) );
             }
             $this->siteBasics['module-run-required'] = false;
-            if ( $this->module->exitStatus() == eZModule::STATUS_RERUN )
+            if ( isset( $moduleResult['rerun_uri'] ) )
             {
-                if ( isset( $moduleResult['rerun_uri'] ) )
-                {
-                    $this->uri = eZURI::instance( $moduleResult['rerun_uri'] );
-                    $this->siteBasics['module-run-required'] = true;
-                }
-                else
-                    eZDebug::writeError( 'No rerun URI specified, cannot continue', 'index.php' );
+                $this->uri = eZURI::instance( $moduleResult['rerun_uri'] );
+                $this->siteBasics['module-run-required'] = true;
             }
 
```

Comment:
Maybe also add something like this to retain the info on inconsistent status and missing rerun uri.

```suggestion
            else if ( $this->module->exitStatus() == eZModule::STATUS_RERUN )
            {
                eZDebug::writeError( 'No rerun URI specified on eZModule::STATUS_RERUN , cannot set URI', 'index.php' );
            }
```

Reply:
Done, thanks!